cmd/dep/testdata/harness_tests/init/gb/case1/final/Gopkg.toml

```diff
@@ -0,0 +1,20 @@
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/carolynvs/deptest"
+  source = "https://github.com/carolynvs/deptest"
```

Comment:
These `source` fields shouldn't be present, because they aren't forks or non-standard repo locations.

It looks like gb always has `pkg.Repository` set... We need to come up some way to tell when we should set `source`. 🤔 @sdboyer I assume that gps already has logic that does that but I'm not sure where or it it's exposed.